### PR TITLE
Added hovered and selected behaviour to dropdown items.

### PR DESCRIPTION
**Pull request by jdhodges**
#579.

### CODE REVIEW

File: src/assets/js/app.js [hunk 1]
```diff
@@ -376,7 +376,7 @@ export default function() {
                 return option.isSelected && option.isPrimary;
             });
 
-        model.overlay.primaryIndicators = model.primaryChart.indicators;
+        model.overlay.primaryIndicators = model.headMenu.primaryIndicators = model.primaryChart.indicators;
```
Comment (djmiley) [inline src/assets/js/app.js]: Is this best practice? (I don't actually know)

File: src/assets/js/app.js [hunk 2]
```diff
@@ -389,7 +389,7 @@ export default function() {
             chartOption.option.on(event.viewChange, onViewChange);
         });
 
-        model.overlay.secondaryIndicators = charts.secondaries;
+        model.overlay.secondaryIndicators = model.headMenu.secondaryIndicators = charts.secondaries;
```
Comment (djmiley) [inline src/assets/js/app.js]: Is this best practice? (I don't actually know)

File: src/assets/js/menu/selectors.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ export default function() {
                     return option.isSelected ? index : null;
                 })
                 .filter(function(option) {
-                    return option;
+                    return option !== null;
```
Comment (djmiley) [inline src/assets/js/menu/selectors.js]: Why this check? When do we get `undefined`?

File: src/assets/js/menu/generator/dropdown.js [hunk 4]
```diff
@@ -63,6 +63,10 @@ export default function() {
         listItemAnchors.append('span')
             .attr('class', 'name');
 
+        listItems.classed('selected', function(d, i) {
+            return model.selected ? model.selected.indexOf(i) > -1 : i === selectedIndex;
```
Comment (djmiley) [inline src/assets/js/menu/generator/dropdown.js]: Can we change `selected` to `selectedIndexes` to make it clear that we are doing this when there may be multiple things selected?

**Comment by djmiley** [comment]
Looks like an effective fix at the moment :)

File: src/assets/js/menu/selectors.js (hunk 3 above)
Comment (jdhodges) [inline src/assets/js/menu/selectors.js]: Your comment made me come up with a much better way of doing this using reduce :smile:

File: src/assets/js/app.js (hunk 1 above)
Comment (jdhodges) [inline src/assets/js/app.js]: I think I'll split this into multiple lines to avoid confusion.

**Comment by djmiley** [comment]
Once that's updated, I'll +1 it.

File: src/assets/js/app.js [hunk 5]
```diff
@@ -389,8 +390,9 @@ export default function() {
             chartOption.option.on(event.viewChange, onViewChange);
         });
 
-        model.overlay.secondaryIndicators = charts.secondaries;
-        // TODO: Remove .remove! (could a secondary chart group component manage this?).
+        model.headMenu.secondaryIndicators = charts.secondaries;
+        model.overlay.secondaryIndicators = model.headMenu.secondaryIndicators;
```
Comment (djmiley) [inline src/assets/js/app.js]: `model.overlay.secondaryIndicators = charts.secondaries`

**Pull request closed without merging**